### PR TITLE
[ARCTIC-683][Flink] fixed the problem that passing a null value to some fields would be given a default value

### LINKED ISSUE #683: [Bug]: When writing data with a null value to the logstore of an arctic table, a non-null default value is assigned to the field
### What happened?

When writing data with a null value to the log-store of an arctic table, a non-null default value is assigned to the field. For example: in the case of a table with schema (int, int, double, float, string), inserting a value of (1, null, null, null, null, 'a') and then reading this data from the log-store of the arctic table will return (1, 0, 0.0, 0.0, 'a')

### Affects Versions

0.3.2

### What engines are you seeing the problem on?

Flink

### How to reproduce

First create a test table with schema (int, int, double, float, string), then write data to the logstore of the arctic table by reading the contents of the kafka topic,such as (1, null, null, null, null, 'a') and finally read the logstore of the arctic table to get the data with the default values (1, 0, 0.0, 0.0, 'a')

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by lklhdu**
fix #683 

## Why are the changes needed?

*When users insert data into the log store of an arctic table via Flink, some fields of the entire data have a null value, but when the data is read from the log store these fields with a null value are given a default value, which should be fixed.*

## Brief change log

  - *Added logic for null value determination to decimal and several other field types*
  - *The Flink 1.12, 1.14, and 1.15 versions are affected.*
  - *resolve lambda serialization error*

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run the test locally before making a pull request

### CODE REVIEW

**Review by YesOrNo828 — COMMENTED** [review]
@lklhdu , Thanks for your contribution. I think adding some tests about null values during the serialization is essential.

File: core/pom.xml [hunk 1]
```diff
@@ -131,6 +131,14 @@
             <scope>test</scope>
         </dependency>
 
+
```
Comment (zstraw) [inline core/pom.xml:134]: don't import flink into core module

File: core/src/main/java/com/netease/arctic/log/LogDataJsonSerialization.java [hunk 2]
```diff
@@ -42,7 +46,12 @@
 
   private transient LogDataToJsonConverters.LogDataToJsonConverter.FormatConverterContext converterContext;
 
-  public LogDataJsonSerialization(Schema schema, LogData.FieldGetterFactory<T> fieldGetterFactory) {
+  public LogDataJsonSerialization(Schema schema, FieldGetterFactory<T> fieldGetterFactory) {
+    this.schema = schema;
+    this.fieldGetterFactory = fieldGetterFactory;
+  }
+
+  public void init() {
```
Comment (zstraw) [inline core/src/main/java/com/netease/arctic/log/LogDataJsonSerialization.java:54]: Please add annotation to remind devs of that the method should be invoked before serialize()

**Review by zstraw — COMMENTED** [review]
I leave some comments, please take a look.

File: core/src/main/java/com/netease/arctic/log/LogDataJsonSerialization.java (hunk 2 above)
Comment (zhoujinsong) [reply core/src/main/java/com/netease/arctic/log/LogDataJsonSerialization.java:54]: Or shall we automatically initialize this variable when using it?

Comment (lklhdu) [reply core/src/main/java/com/netease/arctic/log/LogDataJsonSerialization.java:54]: > 

One way I can think of to automatically initialize this variable is to judge if the logDataToJsonConverter variable has been initialized when the serialize() method is called and initialize the variable if it has not. I wonder if you think this solution would work

Comment (zhoujinsong) [reply core/src/main/java/com/netease/arctic/log/LogDataJsonSerialization.java:54]: Yes, the same as I think about.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**